pkg/controller/cloud/node_controller_test.go

```diff
@@ -672,6 +677,128 @@ func TestNodeProvidedIPAddresses(t *testing.T) {
 	assert.Equal(t, "10.0.0.1", updatedNodes[0].Status.Addresses[0].Address, "Node Addresses not correctly updated")
 }
 
+func Test_reconcileNodeLabels(t *testing.T) {
+	testcases := []struct {
+		name         string
+		existingNode *v1.Node
```

Comment:
`reconcileNodeLabels` actually calls the client so had to use a real node object

Reply:
i just mean the input to the test can be the most specific thing, you can then add the `Labels` into a real node object inside the test (since the rest of the node object is the same within all the tests) - Alternatively, you can make a function to generate the `Node`

Reply:
ah good point, I can do that in a bit